src/main/java/relations/r1cs/FileToR1CS.java

```diff
@@ -0,0 +1,478 @@
+        JSONParser parser = new JSONParser();
+        JSONObject jsonObject;
+        JSONArray primaryInputs = new JSONArray();
+        JSONArray auxInputs = new JSONArray();
+        JSONArray constraintList = new JSONArray();
+
+        try {
+            Object obj = parser.parse(new FileReader(filePath));
+
+            jsonObject = (JSONObject) obj;
+            primaryInputs = (JSONArray) jsonObject.get("primary_input");
+            auxInputs = (JSONArray) jsonObject.get("aux_input");
+            constraintList = (JSONArray) jsonObject.get("constraints");
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        final Assignment<FieldT> oneFullAssignment = new Assignment<>();
+
+        int numInputs = primaryInputs.size();
+        for (int i = 0; i < numInputs; i++) {
+            final BN254aFields.BN254aFr value = new BN254aFields.BN254aFr((String) primaryInputs.get(i));
+            oneFullAssignment.add((FieldT) value);
+        }
+
+        int numAuxiliary = auxInputs.size();
+        for (int i = 0; i < numAuxiliary; i++) {
+            final BN254aFields.BN254aFr value = new BN254aFields.BN254aFr((String) auxInputs.get(i));
+            oneFullAssignment.add((FieldT) value);
+        }
+
+        int numConstraints = constraintList.size();
+        JSONArray[] constraintArray = new JSONArray[numConstraints];
+        for (int i = 0; i < numConstraints; i++) {
+            constraintArray[i] = (JSONArray) constraintList.get(i);
+        }
+
+        final R1CSConstraints<FieldT> constraints = new R1CSConstraints<>();
+
+        for (int i = 0; i < constraintArray.length; i++) {
+
+            JSONObject nextA = (JSONObject) constraintArray[i].get(0);
+            JSONObject nextB = (JSONObject) constraintArray[i].get(1);
+            JSONObject nextC = (JSONObject) constraintArray[i].get(2);
+
+
+            final LinearCombination<FieldT> A = JSONObjectToCombination(nextA);
+            final LinearCombination<FieldT> B = JSONObjectToCombination(nextB);
+            final LinearCombination<FieldT> C = JSONObjectToCombination(nextC);
+
+            constraints.add(new R1CSConstraint<>(A, B, C));
+        }
+
+        final R1CSRelation<FieldT> r1cs = new R1CSRelation<>(constraints, numInputs, numAuxiliary);
+        final Assignment<FieldT> primary = new Assignment<>(oneFullAssignment.subList(0, numInputs));
+        final Assignment<FieldT> auxiliary = new Assignment<>(oneFullAssignment
+                .subList(numInputs, oneFullAssignment.size()));
+
+        assert (r1cs.numInputs() == numInputs);
+        assert (r1cs.numVariables() >= numInputs);
+        assert (r1cs.numVariables() == oneFullAssignment.size());
+        assert (r1cs.numConstraints() == numConstraints);
+        assert (r1cs.isSatisfied(primary, auxiliary));
+
+        return new Tuple3<>(r1cs, primary, auxiliary);
+    }
+
+    public static <FieldT extends AbstractFieldElementExpanded<FieldT>>
```

Comment:
I think perhaps it would be best to split up serial and parallel constructions into their own class. That would divide the size by 2 (and I am still removing duplicated code). If you agree, I would make two classes;
1. `DistributedR1CSFromFile` and 
2. `SerialR1CSFromFile`.

Each would have functions `R1CSFromJSON` and `R1CSFromText` along with a couple of helper functions. In the future, there would also be `R1CSFromBinary`.

Reply:
Done! In next push